realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -900,13 +1041,68 @@ private ErrorCode combineErrorCodes(Map<String, ObjectServerError> errors) {
 
     }
 
+    private interface ErrorCallback {
```

Comment:
The name of this interface should be `PermissionBaseCallback` or similar.
`PermissionsCallback is a ErrorCallback` is odd, isn't it?